src/logexp.jl

```diff
@@ -1,49 +1,41 @@
 ## log
-# 3d
-function Base.log(R::RotationVec)
-    x, y, z = params(R)
-    return RotationVecGenerator(x,y,z)
-end
-
-function Base.log(R::Rotation{3})
-    log(RotationVec(R))
-end
-
-
 # 2d
-function Base.log(R::Angle2d)
-    θ, = params(R)
-    return Angle2dGenerator(θ)
-end
-
-function Base.log(R::Rotation{2})
-    log(Angle2d(R))
-end
-
+Base.log(R::Angle2d) = Angle2dGenerator(R.theta)
+Base.log(R::Rotation{2}) = log(Angle2d(R))
+Base.log(R::RotMatrix{2}) = RotMatrixGenerator(log(Angle2d(R)))
 
-## exp
 # 3d
-function Base.exp(R::RotationVecGenerator)
-    return RotationVec(R.x,R.y,R.z)
-end
-
-function Base.exp(R::RotationGenerator{3})
-    exp(RotationVecGenerator(R))
-end
-
-function Base.exp(R::RotMatrixGenerator{3})
-    RotMatrix(exp(RotationVecGenerator(R)))
+Base.log(R::RotationVec) = RotationVecGenerator(R.sx,R.sy,R.sz)
+Base.log(R::Rotation{3}) = log(RotationVec(R))
+Base.log(R::RotMatrix{3}) = RotMatrixGenerator(log(RotationVec(R)))
+
+# General dimensions
+if VERSION < v"1.7"
+    # This if block is related to this PR.
+    # https://github.com/JuliaLang/julia/pull/40573
+    function Base.log(R::RotMatrix{N}) where N
+        # This will be faster when log(::SMatrix) is implemented in StaticArrays.jl
+        S = SMatrix{N,N}(real(log(Matrix(R))))
+        RotMatrixGenerator((S-S')/2)
+    end
+else
+    function Base.log(R::RotMatrix{N}) where N
+        # This will be faster when log(::SMatrix) is implemented in StaticArrays.jl
+        S = SMatrix{N,N}(log(Matrix(R)))
+        RotMatrixGenerator((S-S')/2)
+    end
```

Comment:
This could also be written this way with less duplication

```suggestion
function Base.log(R::RotMatrix{N}) where N
    # This will be faster when log(::SMatrix) is implemented in StaticArrays.jl
    @static if VERSION < v"1.7"
        # This if block is related to this PR.
        # https://github.com/JuliaLang/julia/pull/40573
        S = SMatrix{N,N}(real(log(Matrix(R))))
    else
        S = SMatrix{N,N}(log(Matrix(R)))
    end
    RotMatrixGenerator((S-S')/2)
end
```

Reply:
Thanks! I did not know the `@static` macro.